### PR TITLE
Added hover effect on sidebar elements

### LINKED ISSUE #886: Hover effect on sidebar elements
#### Nature of issue?

- Existing feature enhancement

#### Details about the bug: 

- Web browser and version: Chrome
- Operating System:  Windows
- Steps to reproduce this bug:

#### Feature enhancement details:
add hover effect on side bar elements
Capture

Earlier there was no hover effect on sidebar elements like "remane" or "delete"

### PR DESCRIPTION
**Pull request by plxity**
I have verified that this pull request:
fixes #886
* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: client/modules/IDE/components/Sidebar.jsx [hunk 1]
```diff
@@ -69,12 +69,12 @@ class Sidebar extends React.Component {
               <InlineSVG src={downArrowUrl} />
             </button>
             <ul className="sidebar__project-options">
-              <li>
+              <li className="sidebar_file">
```
Comment (catarak) [inline client/modules/IDE/components/Sidebar.jsx]: following [BEM](http://getbem.com/) conventions, i would name this class "sidebar__project-option"

File: client/modules/IDE/components/FileNode.jsx [hunk 2]
```diff
@@ -162,7 +162,7 @@ export class FileNode extends React.Component {
                     {(() => { // eslint-disable-line
                       if (this.props.fileType === 'folder') {
                         return (
-                          <li>
+                          <li className="sidebar_file">
```
Comment (catarak) [inline client/modules/IDE/components/FileNode.jsx]: following BEM conventions, i would name this class "sidebar__file-item-option"

File: client/styles/components/_sidebar.scss [hunk 3]
```diff
@@ -286,8 +285,42 @@
   background-color: transparent;
   border: none;
 }
+.sidebar_file{
```
Comment (catarak) [inline client/styles/components/_sidebar.scss]: since this is using the same code as the Nav, maybe this should be moved into a SASS placeholder or mixin?

**Comment by plxity** [comment]
@catarak I have made the following changes. Please review it again.

**Comment by catarak** [comment]
could you remove changes to `package-lock.json`? also, you didn't move the styles to a mixin or placeholder even though they are duplicate styles.

**Comment by plxity** [comment]
@catarak Okay, I will remove changes from `package-lock.json`.
Actually, There is a difference in the styles, I have removed a line or two to make it look proper.

**Comment by catarak** [comment]
you can still use a placeholder or mixin! you can extend it and then override those styles.

**Comment by plxity** [comment]
@catarak Okay, I will fix that.

**Comment by plxity** [comment]
@catarak  I have used a different approach now. and have implemented the changes. Can you please review this? :)
Thank you.

**Comment by catarak** [comment]
here's some stuff i'm noticing: 
Screen Shot 2019-04-17 at 2 49 26 PM
the highlight doesn't reach the full width/height of each item
Screen Shot 2019-04-17 at 2 49 37 PM
it looks a little weird without the border radius. i think the style should match the navigation dropdown
Screen Shot 2019-04-17 at 2 49 44 PM
when you hover over the first item for a folder dropdown, it doesn't highlight

i think that this should use the same styles as the navigation dropdown, so i think this should be moved into a placeholder, rather than repeating the same code twice.

**Comment by plxity** [comment]
@catarak Okay, I will look into it once. and make the following changes :)

**Pull request merged**